docs/examples/Serialization Example.ipynb

```diff
@@ -591,11 +604,261 @@
+      "node attributes: ['id', 'u_rated']\n",
+      "source attributes: ['id', 'node', 'status', 'u_ref', 'sk']\n",
+      "sym_load attributes: ['id', 'node', 'status', 'type', 'p_specified', 'q_specified']\n",
+      "line attributes: ['id', 'from_node', 'to_node', 'from_status', 'to_status', 'r1', 'x1', 'c1', 'tan1', 'i_n']\n",
+      "-------node data-------\n"
+     ]
+    },
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>id</th>\n",
+       "      <th>u_rated</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>1</td>\n",
+       "      <td>10500.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>2</td>\n",
+       "      <td>10500.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>3</td>\n",
+       "      <td>10500.0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   id  u_rated\n",
+       "0   1  10500.0\n",
+       "1   2  10500.0\n",
+       "2   3  10500.0"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "dataset = json_deserialize(\n",
+    "    data,\n",
+    "    data_filter={\n",
+    "        ComponentType.node: None,  # nodes in a row-based data format\n",
+    "        ComponentType.source: [\"id\", \"node\", \"status\", \"u_ref\", \"sk\"],  # only specific attributes\n",
+    "        ComponentType.sym_load: ComponentAttributeFilterOptions.everything,  # all attributes as columns\n",
+    "        ComponentType.line: ComponentAttributeFilterOptions.relevant,  # only attributes that are not null/nan\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "print(\"components:\", list(dataset.keys()))\n",
+    "print(\"node attributes:\", list(dataset[ComponentType.node].dtype.names))\n",
+    "print(\"source attributes:\", list(dataset[ComponentType.source].keys()))\n",
+    "print(\"sym_load attributes:\", list(dataset[ComponentType.sym_load].keys()))\n",
+    "print(\"line attributes:\", list(dataset[ComponentType.line].keys()))\n",
+    "print(\"-------node data-------\")\n",
+    "display(DataFrame(dataset[ComponentType.node]))"
```

Comment:
Nitpick: Maybe display the `source` instad? Just to illustrate columnar?

Reply:
removed entirely